src/inmanta/server/services/environment_metrics_service.py

```diff
@@ -194,3 +195,29 @@ def create_metrics_timer(metric_values_timer: Sequence[MetricValueTimer], timest
                 "Verify the load on the Database and the available connection pool size.",
                 COLLECTION_INTERVAL_IN_SEC,
             )
+
+
+class ResourceCountMetricsCollector(MetricsCollector):
+    """
+    This Metric will track the number of resources (grouped by resources state).
+    """
+
+    def get_metric_name(self) -> str:
+        return "resource_count"
+
+    def get_metric_type(self) -> MetricType:
+        return MetricType.GAUGE
+
+    async def get_metric_value(
+        self, start_interval: datetime, end_interval: datetime, connection: Optional[asyncpg.connection.Connection]
+    ) -> Sequence[MetricValue]:
+        query: str = """
+            SELECT status,count(*)
+            FROM resource
+            GROUP BY status
+        """
+        metric_values: List[MetricValue] = []
+        result: Sequence[asyncpg.Record] = await connection.fetch(query)
```

Comment:
The type annotation for connection is optional, but no is None check is done here. I think we can just make it a mandatory argument. Do you also run mypy on this?